### PR TITLE
feat(datepicker): add possibility for a format function to Datepicker

### PR DESCRIPTION
**Pull request by Calamari**
## Proposed changes
This is described in #151. In short: This adds the ability to add a formatting function as format option to datepicker instance, which allows things like this:

```js
M.Datepicker.init(elems, {
  format: date => new Intl.DateTimeFormat(locale).format(date)
});
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist:

- [x] I have read the **[CONTRIBUTING document](https://github.com/materializecss/materialize/blob/v1-dev/CONTRIBUTING.md)**.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by roiLeo — APPROVED** [review]
lgtm
#151 related

**Review by Smankusors — APPROVED** [review]
LGTM. Well, the check failing because there's a problem with your commit messages

What do you think about these commit messages? @ChildishGiant

**Comment by ChildishGiant** [comment]
It's just an issue of the case, which doesn't overly matter. I might change the action to only give a warning as it's not a massive deal if the commits aren't the right style.

**Comment by ray007** [comment]
Should a format-function not have a `format` parameter?

**Comment by Smankusors** [comment]
> Should a format-function not have a `format` parameter?

hmm wait, I don't get it. What do you mean?

**Comment by ray007** [comment]
> 
> 
> > Should a format-function not have a `format` parameter?
> 
> hmm wait, I don't get it. What do you mean?

Haven't yet looked at your pull request, but I guess it's probably already possible:
``` javascript
M.Datepicker.init(elems, {
  format: (date, fmt) => new Intl.DateTimeFormat(locale, fmt).format(date)
});
```

**Comment by Calamari** [comment]
@ray007 and what value should the `fmt` parameter contain?

If you add a format function, why should it not be possible for that function to know how to format itself?

**Comment by Calamari** [comment]
So, I finally fixed those commit linting problems :-)

File: pug/page-contents/pickers_content.html [hunk 1]
```diff
@@ -57,9 +57,9 @@ <h5>Options</h5>
               </tr>
               <tr>
                 <td>format</td>
-                <td>String</td>
+                <td>String || Function</td>
                 <td>'mmm dd, yyyy'</td>
-                <td>The date output format for the input field value.</td>
+                <td>The date output format for the input field value or a function taking the date and outputitng the formatted date string.</td>
```
Comment (DanielRuf) [inline pug/page-contents/pickers_content.html:62]:
```suggestion
                <td>The date output format for the input field value or a function taking the date and outputting the formatted date string.</td>
```

**Review by DanielRuf — APPROVED** [review]
Just a small typo, rest looks good to me.

**Comment by Calamari** [comment]
So, let's see if I can finally pass the bouncer of a commit linter.

**Comment by Calamari** [comment]
Anyone like to approve the workflow? :-)

**Comment by DanielRuf** [comment]
> Anyone like to approve the workflow? :-)

As far as I can see it was already approved and successfully finished (2 successful checks).

grafik

**Comment by Smankusors** [comment]
> As far as I can see it was already approved and successfully finished (2 successful checks).

Oh yeah, I did approve it.

Hmm.... I assume the issue that mentioned by @ray007 is already solved (there is no reply since a month).  Merging it now.

**Comment by Smankusors** [comment]
Anyway, I sent an invitation to you to join the @materializecss/members team. Welcome 🎉

**Comment by Calamari** [comment]
To me it said, it waited for approval for the workflows, that's why I wrote again. :-)
Thanks :-)

**Comment by DanielRuf** [comment]
> To me it said, it waited for approval for the workflows, that's why I wrote again. :-)

I have probably manually triggered them a few days or weeks ago.

**Pull request merged**